src/psyclone/psyir/transformations/replace_reference_by_literal_trans.py

```diff
@@ -132,35 +225,52 @@ def _replace_bounds(
 
     # ------------------------------------------------------------------------
     def apply(self, node: Routine, options=None):
+        """Applies the transformation to a Routine node:
+        * First update a dictionary (param_table) with the Literal of constant
+        (parameter) symbol from node.parent symbol_table, and from
+        node.symbol_table.
+        * Second, use this updated param_table to replace reference in node
+        psyir_tree with the corresponsing Literal.
+        * Third, use this updated param_table to replace reference in node
+        symbol_table DataSymbol array's dimensions with the corresponsing
+        Literal.
+
```

Comment:
Please add a note here to say that if an update cannot be performed for any reason then the substitution is skipped and a comment added to the generated code.